src/refract/JsonSchema.cc

```diff
@@ -323,28 +323,29 @@ namespace
     void renderProperty(ObjectSchema& s, const ExtendElement& e, TypeAttributes options);
     void renderProperty(ObjectSchema& s, const IElement& e, TypeAttributes options);
 
-    so::Object& renderSchema(so::Object& schema, const ObjectElement& e, TypeAttributes options);
-    so::Object& renderSchema(so::Object& schema, const ArrayElement& e, TypeAttributes options);
-    so::Object& renderSchema(so::Object& schema, const EnumElement& e, TypeAttributes options);
-    so::Object& renderSchema(so::Object& schema, const NullElement& e, TypeAttributes options);
-    so::Object& renderSchema(so::Object& schema, const StringElement& e, TypeAttributes options);
-    so::Object& renderSchema(so::Object& schema, const NumberElement& e, TypeAttributes options);
-    so::Object& renderSchema(so::Object& schema, const BooleanElement& e, TypeAttributes options);
-    so::Object& renderSchema(so::Object& schema, const ExtendElement& e, TypeAttributes options);
+    so::Object& renderSchemaSpecific(so::Object& schema, const ObjectElement& e, TypeAttributes options);
+    so::Object& renderSchemaSpecific(so::Object& schema, const ArrayElement& e, TypeAttributes options);
+    so::Object& renderSchemaSpecific(so::Object& schema, const EnumElement& e, TypeAttributes options);
+    so::Object& renderSchemaSpecific(so::Object& schema, const NullElement& e, TypeAttributes options);
+    so::Object& renderSchemaSpecific(so::Object& schema, const StringElement& e, TypeAttributes options);
+    so::Object& renderSchemaSpecific(so::Object& schema, const NumberElement& e, TypeAttributes options);
+    so::Object& renderSchemaSpecific(so::Object& schema, const BooleanElement& e, TypeAttributes options);
+    so::Object& renderSchemaSpecific(so::Object& schema, const ExtendElement& e, TypeAttributes options);
+    so::Object& renderSchemaSpecific(so::Object& schema, const RefElement& e, TypeAttributes options);
     so::Object& renderSchema(so::Object& schema, const IElement& e, TypeAttributes options);
 
     template <typename T>
     void renderProperty(ObjectSchema&, const T& e, so::Object&, TypeAttributes)
     {
         LOG(error) << "invalid property element: " << e.element();
-        assert(false);
+        abort();
```

Comment:
Can you explain the rationale behind changing assert to abort please, not sure I understand the reason

Reply:
Here we want to avoid continuing the program flow because we should never arrive there. Specifically, bug #709 caused a SIGSEGV by overflowing the stack through unchecked recursion - with an abort here, there would be the much nicer SIGABORT.

Reply:
I see, perhaps it would be better to avoid exiting the process though? Perhaps throwing std::exception would be better, the drafter_parse/validate function could catch and return appropriate error to caller?